torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -1425,10 +1427,19 @@ class ShapePropagator {
       at::optional<IValue> maybe_dtype_option = node->get(attr::dtype);
       if (!maybe_dtype_option)
         return {};
-      auto dtype =
-          (maybe_dtype_option->isNone() ? at::kDouble
-                                        : maybe_dtype_option->toScalarType());
 
+      c10::optional<c10::ScalarType> in_dtype;
+      if (auto in_tt = node->input(0)->type()->cast<TensorType>()) {
+        in_dtype = in_tt->scalarType();
+      } else {
+        in_dtype = c10::kDouble;
```

Comment:
I don't think this is right, I think we should be using the current default dtype probably

Reply:
the current logic breaks quantization tests. Namely, we infer `kDouble` for `torch.zeros_like(min_val)` even though we should use `min_val's dtype which is `kFloat` (in this case we infer `min_val`'s dtype correctly) or if it's not known, we shouldn't probably try to infer `torch.zeros_like`'s dtype, so when we try running torch.min with representative double and float values it complains.

```
min_val = torch.min(min_val, torch.zeros_like(min_val))
```

Reply:
when is the `in_dtype = c10::kDouble` called then ? i still think it should be `c10::current_default_dtype`